core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java

```diff
@@ -53,10 +54,51 @@ public TableSink(TableSinkConfig tableSinkConfig) {
   }
 
   @Override
-  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
     Preconditions.checkArgument(!Strings.isNullOrEmpty(tableSinkConfig.getRowField()),
                                 "Row field must be given as a property.");
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      validateInputAndOutputSchemaFields(inputSchema);
+
+      boolean rowFieldPresent = false;
+      for (Schema.Field field : inputSchema.getFields()) {
+        // simple type check for fields
+        if (!field.getSchema().getType().isSimpleType()) {
+          throw new IllegalArgumentException(
+            String.format("Field '%s' is not of simple type, All fields for table sink should of simple type",
+                          field.getName()));
+        }
+        // check if this field is the row key field
+        if (field.getName().equals(tableSinkConfig.getRowField())) {
+          rowFieldPresent = true;
+        }
+      }
+
+      if (!rowFieldPresent) {
+        throw new IllegalArgumentException(
+          String.format("Row field key : '%s' is not present in the input schema",  tableSinkConfig.getRowField()));
+      }
+    }
+  }
+
+  private void validateInputAndOutputSchemaFields(Schema inputSchema) {
+    try {
+      // check if input schema contains all the fields expected in the output schema (if output schema is provided)
+      if (tableSinkConfig.getSchemaStr() != null) {
+        Schema outputSchema = Schema.parseJson(tableSinkConfig.getSchemaStr());
+        for (Schema.Field field : outputSchema.getFields()) {
+          if (inputSchema.getField(field.getName()) == null) {
+            throw new IllegalArgumentException(
+              String.format("Field '%s' is present in output schema but not present in input schema",
+                            field.getName()));
+          }
```

Comment:
seems like a common method, could put it in the common package somewhere as a `isSubset` method or something like that